### PR TITLE
feat: Add description to classes

### LINKED ISSUE #363: Add ability to put description attributes on classes, and show those descriptions on the admin list and edit pages.

### PR DESCRIPTION
**Pull request by meghanmae**
Adding a description for classes

Fixes #363 

Images from 6409a64613cbe5c3c5fd3f8a6d5d0e8a843d0e20
image
image

Images from e99895377936c5de056dc83da1a3cf5bf2f7be1e
~[image](https://github.com/user-attachments/assets/de7eeba3-48ba-4798-b9e2-ef86e4c4d274)~
~[image](https://github.com/user-attachments/assets/a1c79217-c4ce-4df6-93da-d4fd4c15e17c)~

### CODE REVIEW

File: src/coalesce-vue/src/metadata.ts [hunk 1]
```diff
@@ -148,6 +151,7 @@ export interface ModelType extends CustomReferenceTypeBase, ApiRoutedType {
 export interface DataSourceType extends Metadata {
   readonly type: "dataSource";
   readonly isDefault?: true;
+  readonly description?: string;
```
Comment (meghanmae) [inline src/coalesce-vue/src/metadata.ts:154]: Unsure about this change - added to fix a test failure.

Comment (ascott18) [reply src/coalesce-vue/src/metadata.ts:154]: Which test is failing?

File: src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue [hunk 2]
```diff
@@ -56,27 +51,35 @@
...
-        v-model="list.$params.search"
-        density="comfortable"
-        single-line
-        clearable
-      ></v-text-field>
-
-      <v-divider class="hidden-xs-only mx-4 my-0" vertical></v-divider>
-
-      <c-list-filters :list="list" />
+  <v-sheet :color="color">
```
Comment (ascott18) [inline src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue:56]: I'm not sure about putting this in the toolbar - especially with the amount of dismantling of the actual toolbar structure that it required.

I think it should go in the body of the card, at the top:
image
image

``` scss
.c-admin-table--description {
  font-size: 12px;
  margin: 4px 16px;
  opacity: 0.9;
}
```

Likewise for the editor

**Review by ascott18 — CHANGES_REQUESTED** [review]

File: src/coalesce-vue/src/metadata.ts (hunk 1 above)
Comment (meghanmae) [reply src/coalesce-vue/src/metadata.ts:154]: `TargetClassesFullGenerationTest.VueOutputCompiles`
Specifically the protected method to check TS compilation `TargetClassesFullGenerationTest.AssertVueSuiteTypescriptOutputCompiles` with the message:
:../IntelliTect.Coalesce.CodeGeneration.Tests/out/.NETCoreApp,Version=v8.0/VueSuite/src/metadata.g.ts(330,7): error TS2353: Object literal may only specify known properties, and 'description' does not exist in type 'DataSourceType'.

Comment (meghanmae) [reply src/coalesce-vue/src/metadata.ts:154]: Ah - this is from the empty strings, I think. So with the change from the comment below this code should be removed. Didn't totally understand how these tests worked, but that's a clearer picture now. 
https://github.com/IntelliTect/Coalesce/pull/419#discussion_r1714034389

Comment (ascott18) [reply src/coalesce-vue/src/metadata.ts:154]: Oh, yeah, that's fine.

Comment (ascott18) [reply src/coalesce-vue/src/metadata.ts]: You can keep it - someone could still put a description attribute on a data source and we don't want that to have type errors (even if we don't currently do anything with hypothetical descriptions on datasources).

File: src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue [hunk 3]
```diff
@@ -4,6 +4,7 @@
     class="c-admin-table-toolbar"
     density="comfortable"
     :color="color"
+    elevation="4"
```
Comment (meghanmae) [inline src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue:7]: I can move this back - it just seemed like it was in a weird spot.

File: src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue (hunk 2 above)
Comment (meghanmae) [reply src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue]: That sounds good, I wasn't really sure about where to put this. I started with it in the body, but it looked a bit weird (my font was default size, though). I think the smaller text helps.

File: src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue (hunk 3 above)
Comment (ascott18) [reply src/coalesce-vue-vuetify3/src/components/admin/c-admin-table-toolbar.vue:7]: Probably fine - direct usages of c-admin-table-toolbar are rare.

File: src/coalesce-vue-vuetify3/src/components/admin/c-admin-editor.vue [hunk 4]
```diff
@@ -54,6 +54,11 @@
       </v-btn>
     </v-toolbar>
 
+    <div v-if="metadata.description" class="c-admin-table--description">
```
Comment (ascott18) [inline src/coalesce-vue-vuetify3/src/components/admin/c-admin-editor.vue:57]: Class name doesn't match this component. Could use a slightly more generic class name if we want to reuse the class on both components.

File: docs/modeling/model-components/attributes.md [hunk 5]
```diff
@@ -20,6 +20,9 @@ The displayed name and description of a property, as well as the order in which
 
 The displayed name of a property can also be set via the `[DisplayName]` attribute.
 
+### [Description]
+The description of a type, such as a class, property, method, or parameter.
```
Comment (ascott18) [inline docs/modeling/model-components/attributes.md:24]:
```suggestion
The description of a type or member, such as a class, property, method, or parameter.
```

File: src/coalesce-vue-vuetify3/src/components/admin/c-admin-editor.vue (hunk 4 above)
Comment (meghanmae) [reply src/coalesce-vue-vuetify3/src/components/admin/c-admin-editor.vue]: Hm, not really sure what to name it. Looking through the rest of the names it doesn't look like you ever really use the same CSS class across components like this. Do you have a preference? I could also move it to the c-admin-table-page and name it c-admin-description? Also seems a bit weird though.

Comment (ascott18) [reply src/coalesce-vue-vuetify3/src/components/admin/c-admin-editor.vue]: c-admin-page--description

**Pull request merged**